File: python/rc4.py
# coding=utf-8

DEFAULT_KEY = "\x59\xf3\x02\xc3\x25\x9a\x82\x30\x0b\xbb\x25\x7f\x7e\x3b\xd2\xdc"

def rc4(data, key=DEFAULT_KEY, skip=1024):
    x = 0
    box = range(256)
    
	

    x = 0
    for i in range(256):
        x = (x + box[i] + ord(key[i % len(key)])) % 256
        tmp = box[i]
        tmp2 = box[x]
        box[i] = box[x]
        box[x] = tmp
    

    x = 0
    y = 0
    out = []
    if skip > 0:
        for i in range(skip):
            x = (x + 1) % 256
            y = (y + box[x]) % 256
            box[x], box[y] = box[y], box[x]
	
    for char in data:
        x = (x + 1) % 256
        y = (y + box[x]) % 256
        box[x], box[y] = box[y], box[x]
        k = box[(box[x] + box[y]) % 256]
        print k
        out.append(chr(ord(char) ^ k))

    return ''.join(out)

if __name__ == '__main__':
    # handle input file or stream
    import sys
    
    rc4("梯子网@91waijiao", "12345678", 0)

File: php/rc4.php
<?php

/*
 * rc4加密算法
 * $pwd 密钥
 * $data 要加密的数据
 */

function rc4 ($pwd, $data)
{
    $key[] ="";
    $box[] ="";
    $cipher = ""; 

    $pwd_length = strlen($pwd);
    $data_length = strlen($data);

        for ($i = 0; $i < 256; $i++)
                    {
                                    $key[$i] = ord($pwd[$i % $pwd_length]);
                                            $box[$i] = $i;
        }


    for ($j = $i = 0; $i < 256; $i++)
    {
        $j = ($j + $box[$i] + $key[$i]) % 256;
        $tmp = $box[$i];
        $box[$i] = $box[$j];
        $box[$j] = $tmp;

        //echo $i.":".$box[$i]." ".$j.":".$box[$j]."\n";
    }
	

    //print_r($box);
   // echo implode($box, ' '), "\n";

    for ($a = $j = $i = 0; $i < $data_length; $i++)
    {
        $a = ($a + 1) % 256;
        $j = ($j + $box[$a]) % 256;

        $tmp = $box[$a];
        $box[$a] = $box[$j];
        $box[$j] = $tmp;

        $k = $box[(($box[$a] + $box[$j]) % 256)];
        echo $k . "\n";

        $cipher .= chr(ord($data[$i]) ^ $k);
    }    
    return $cipher;
}


    
    $key = "12345678";
    $data = "梯子网@91waijiao";
    rc4($key, $data);

